rlog/DRAFT-fiat-shamir.mdx

```diff
@@ -0,0 +1,183 @@
+categories: research
+
+
+toc_min_heading_level: 2
+toc_max_heading_level: 5
+---
+
+In this post, provide a discussion on a common technique used to convert interactive protocols to their noninteractive variant.
+<!--truncate-->
+
+## Introduction
+
+The set of interactive protocols form a class of protocols that consist of communication between two parties: the Prover and the Verifier.
+The Prover tries to convince the Verifier of a given claim.
+For example, the Prover may want to convince the Verifier that she owns a specific UTXO;
+that is, the Prover possesses the ability to spend the UTXO.
+In many instances, there is information that the Prover does not wish to reveal to the Verifier.
+In our example, it is critical that the Prover does not provide the Verifier with the spending key associated to her UTXO.
+In addition to the Prover's claim and secret data, there is additional data, public parameters, that the claimed statement is expressed in terms of.
+The public parameters can be thought of as the basis for all similar claims.
+
+In an interactive protocol, the Prover and the Verifier are in active communication.
+Specifically, the Prover and the Verifier exchange messages so that the Verifier can validate the Prover's claim.
+However, this communication is not practical for many applications.
+It is necessary that any party can verify the Prover's claim in decentralized systems.
+It is impractical for the Prover to be in active communication with a large number of verifying parties.
+Instead, it is desirable for the Prover to generate a proof on their own that can convince any party.
+To achieve this, it is necessary for the Prover to generate the Verifier's messages in such a way that the Prover cannot manipulate the Verifier's messages for her benefit.
+The [Fiat-Shamir heuristic](https://dl.acm.org/doi/10.5555/36664.36676) is used for this purpose.
+
+## Hash Functions
+Cryptographic hash functions serve as the backbone to the Fiat-Shamir heuristic.
+A hash function, $\mathsf{Hash}$, is a special function that takes in an arbitrary binary string and outputs a binary string of a predetermined fixed length.
+Specifically,
+$\mathsf{Hash} : \{0,1\}^* \rightarrow \{0,1\}^n$.
+
+A cryptographic hash function satisfies the following properties:
+- **Succinct**: The hash function should be easy to compute; the hash $\mathsf{Hash}({\bf{b}})$ can be efficiently computed for any binary string ${\bf{b}}$.
+- **Preimage Resistance**: It should be impossible to work backwards given the output of a hash function. Let ${\bf{y}}$ be a binary string of length $n$.
+    It should be 'impossible' to find some binary string ${\bf{x}}$ so that ${\bf{y}} = \mathsf{Hash}({\bf{x}})$. 
+- **Collision Resistance**: It should be difficult to find two strings that hash to the same value. It should be 'impossible' to find two binary strings ${\bf{x}_1}$ and ${\bf{x}_2}$ so that $\mathsf{Hash}({\bf{x}_1}) = \mathsf{Hash}({\bf{x}_2}).$
```

Comment:
```suggestion
- **Collision Resistance**: It should be difficult to find two strings that hash to the same value. It should be computationally infeasible to find two binary strings ${\bf{x}_1}$ and ${\bf{x}_2}$ so that $\mathsf{Hash}({\bf{x}_1}) = \mathsf{Hash}({\bf{x}_2}).$
```